### PR TITLE
Antd typescript support

### PR DESCRIPTION
**Pull request by akki-jat**
for more info look #250

### CODE REVIEW

File: types/tree-select.d.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+   * @default false
+   * @type boolean
+   */
+  treeCheckStrictly: boolean;
+
+  /**
+   * Data of the treeNodes, manual construction work is no longer needed
+   * if this property has been set(ensure the Uniqueness of each value)
+   * @default []
+   * @type aobject[]
```
Comment (wangxueliang) [inline types/tree-select.d.ts]: aobject => object[]

Comment (akki-jat) [reply types/tree-select.d.ts]: @wangxueliang quick to find mistake 😜

File: types/auto-complete.d.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  /**
+   * backfill selected item the input when using keyboard
+   * @default false
+   * @type boolean
+   */
+  backfill: boolean;
+
+  /**
+   * Data source for autocomplete
+   * @type slot | { value: String, text: String }
```
Comment (tangjinzhou) [inline types/auto-complete.d.ts]: * @type slot | Array<{ value: String, text: String }>  ?

File: types/date-picker/range-picker.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+   * to set default date
+   * @type [Moment, Moment]
+   */
+  defaultValue: [Moment, Moment];
+
+  /**
+   * to set default picker date
+   * @type Moment
+   */
+  defaultPickerValue: Moment;
```
Comment (tangjinzhou) [inline types/date-picker/range-picker.ts]: defaultPickerValue: [Moment, Moment];
Sorry, the document has an error

File: types/date-picker/range-picker.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+   * @default TimePicker Options
+   * @type object | boolean
+   */
+  showTime: object | boolean;
+
+  /**
+   * to set date
+   * @type Moment
+   */
+  value: Moment;
```
Comment (tangjinzhou) [inline types/date-picker/range-picker.ts]: value: [Moment, Moment]

File: types/dropdown/dropdown-button.d.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+   * @type Menu
+   */
+  overlay: Menu;
+
+  /**
+   * placement of pop menu: bottomLeft bottomCenter bottomRight topLeft topCenter topRight
+   * @default 'bottomLeft'
+   * @type string
+   */
+  plcement: string;
```
Comment (tangjinzhou) [inline types/dropdown/dropdown-button.d.ts]: plcement => placement

File: types/message.d.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+  /**
+   * distance from top
+   * @default '24px'
+   * @type string
+   */
+  top?: string;
+}
+
+export declare class Message {
```
Comment (tangjinzhou) [inline types/message.d.ts]: add message.loading

File: types/pagination.d.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+   * @type boolean
+   */
+  hideOnSinglePage: boolean;
+
+  /**
+   * determine whether pageSize can be changed
+   * @default false
+   * @type boolean
+   */
+  showSizeChanger: Boolean;
```
Comment (tangjinzhou) [inline types/pagination.d.ts]: Boolean boolean  ?

File: types/radio/radio-button.d.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+// Definitions: https://github.com/vueComponent/ant-design-vue/types
+
+import { Radio } from "./radio";
+
+export declare class RadioButton extends Radio {
+  /**
+   * Type of radio button
+   * @type string
+   */
+  type: String;
```
Comment (tangjinzhou) [inline types/radio/radio-button.d.ts]: I am not sure to support capitalization.

File: types/transfer.d.ts [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+export declare class Transfer extends AntdComponent {
+  /**
+   * Used for setting the source data. The elements that are part of this array will be present the left column.
+   * Except the elements whose keys are included in targetKeys prop.
+   * @default []
+   * @type TransferItem
+   */
+  dataSource: TransferItem;
```
Comment (tangjinzhou) [inline types/transfer.d.ts]: Array<TransferItem>

File: types/tree-node.d.ts [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+  isLeaf: boolean;
+
+  /**
+   * Key
+   * @description Required property, should be unique in the tree
+   * (In tree: Used with (default)ExpandedKeys / (default)CheckedKeys / (default)SelectedKeys)
+   * @default internal calculated position of treeNode or undefined
+   * @type string
+   */
+  key: string;
```
Comment (tangjinzhou) [inline types/tree-node.d.ts]: support number

File: types/tree-select.d.ts [hunk 11]
```diff
@@ -0,0 +1,213 @@
+// Type definitions for Ant Design Vue 1.2.2
+// Project: https://github.com/vueComponent/ant-design-vue
+// Definitions by: akki-jat <https://github.com/akki-jat>
+// Definitions: https://github.com/vueComponent/ant-design-vue/types
+
+import { AntdComponent } from './component';
+import { TreeNode } from './tree-node';
+
+export interface TreeData {
+  key: string;
```
Comment (tangjinzhou) [inline types/tree-select.d.ts]: support number

File: types/tree-select.d.ts [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+  value: string;
+  label: any;
+  children: any;
+  disabled?: boolean;
+  disableCheckbox?: boolean;
+  selectable?: boolean;
+}
+
+export declare class TreeSelect extends AntdComponent {
```
Comment (tangjinzhou) [inline types/tree-select.d.ts]: add new props `treeExpandedKeys` in v1.2.3

File: types/tree/dictionary-tree.d.ts [hunk 13]
```diff
@@ -0,0 +1,15 @@
+// Type definitions for Ant Design Vue 1.2.2
+// Project: https://github.com/vueComponent/ant-design-vue
+// Definitions by: akki-jat <https://github.com/akki-jat>
+// Definitions: https://github.com/vueComponent/ant-design-vue/types
+
+import { AntdComponent } from "../component";
+
+export declare class DictionaryTree extends AntdComponent {
```
Comment (tangjinzhou) [inline types/tree/dictionary-tree.d.ts]: shoule extends `Tree` props

File: types/tree/tree.d.ts [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+   * (PS: When this specifies the key of a treeNode which is also a parent treeNode,
+   * all the children treeNodes of will be checked; and vice versa,
+   * when it specifies the key of a treeNode which is a child treeNode,
+   * its parent treeNode will also be checked. When checkable and checkStrictly is true,
+   * its object has checked and halfChecked property. Regardless of whether the child or parent treeNode is checked,
+   * they won't impact each other.
+   * @default []
+   * @type string[] | { checked: string[]; halfChecked: string[] }
+   */
+  checkedKeys: string[] | { checked: string[]; halfChecked: string[] };
```
Comment (tangjinzhou) [inline types/tree/tree.d.ts]: xxxKeys should support number[]
new feature at 1.2.3

File: types/affix.d.ts [hunk 15]
```diff
@@ -0,0 +1,28 @@
+// Type definitions for Ant Design Vue 1.2.2
```
Comment (tangjinzhou) [inline types/affix.d.ts]: I think we don't need to add version information.

File: types/table/table.d.ts [hunk 16]
```diff
@@ -0,0 +1,294 @@
...
+   * Table title renderer
+   * @type Function | ScopedSlot
+   */
+  title: Function | ScopedSlot;
+
+  /**
+   * Set props on per header row
+   * @type Function
+   */
+  customHeaderRow: (column: any, index: number) => any;
```
[the remark this answers is not in the document]
Comment (akki-jat) [reply types/table/table.d.ts]: @tangjinzhou  Yes we can, but it is very hard to find type for customXXX. So, can we discuss about this?

File: types/dropdown/dropdown-button.d.ts [hunk 17]
```diff
@@ -24,7 +23,13 @@ export declare class DropdownButton extends AntdComponent {
    * @default 'bottomLeft'
    * @type string
    */
-  plcement: string;
+  plcement:
```
[the remark this answers is not in the document]
Comment (akki-jat) [reply types/dropdown/dropdown-button.d.ts]: Is here something wrong?

File: types/table/table.d.ts (hunk 16 above)
[the remark this answers is not in the document]
Comment (akki-jat) [reply types/table/table.d.ts]: is this return type applicable on all customXXX (customRow customHeaderRow customCell customHeaderCel)?l

[the remark this answers is not in the document]
Comment (akki-jat) [reply types/table/table.d.ts]: check now

File: types/dropdown/dropdown-button.d.ts (hunk 17 above)
[the remark this answers is not in the document]
Comment (tangjinzhou) [reply types/dropdown/dropdown-button.d.ts]: Spelling mistake

[the remark this answers is not in the document]
Comment (akki-jat) [reply types/dropdown/dropdown-button.d.ts]: sorry my bad

**Pull request merged**